### ISSUE TITLE
proposal: include this lib as a part of current kalliope project

### ISSUE THREAD

**Issue by Sispheor** [frnmst/kalliope-rest#1]
Hi, 
What do you think about integrating your work into kalliope project?

It would be cool if user that have installed kalliope from pip could use this lib.
And also, we planned to refactor the current integrated GUI with something more python based.

What do you think? I don't know if it possible to place all of his in a dedicated folder so the user could then import kalliope.lib or something like that.

**Comment by frnmst** [comment]
> Hi,
> What do you think about integrating your work into kalliope project?
> 
> It would be cool if user that have installed kalliope from pip could use this lib.
> And also, we planned to refactor the current integrated GUI with something more python based.

Hello,
It's ok with me however I'm not sure what you mean by the current integrated GUI. If you mean CLI  instead of GUI then you probably mean this: https://github.com/kalliope-project/kalliope/blob/b3f21befaaa1b51f81ff8985de81cd48951e1a13/Docs/kalliope_cli.md
and this: https://github.com/kalliope-project/kalliope/blob/be55beadd8e880b83f52e16fd7f92d3c4cbcf060/kalliope/core/ShellGui.py

> What do you think? I don't know if it possible to place all of his in a dedicated folder so the user could then import kalliope.lib or something like that.

That might be possible with some code refactoring...
These are the future plans for this repository: https://github.com/frnmst/todo/blob/master/todo/kalliope-rest.md . For the moment, installation can only be done with: `# make install`. As you can see there quite a lot of things to do.

**Comment by Sispheor** [comment]
Yes I mean the ShellGUI. We plan to remove / refactor it to use api call or lib like the one you made.
Anyway, this is another issue. We'll see this part later.

An so yes, the idea would be to have the lib available when installing Kalliope from pip.
And so the user could then do a "from kalliope import sdk".
The advantage is that he lib will grow in the same time as Kalliope and its API.

Up to you, if it's not a too big task, you can try to 
- implement your lib in a dedicated folder called `sdk` inside Kalliope. Verify that you can import it once Kalliope is reinstalled via setup.py.
- implement the CLI by adding a flag into the current kalliope entry point. Like `kalliope cli <command>`

Just a couple notes:
- the code must have unit tests
- support for python 2 is still mandatory until Raspbian move to Python 3 by default

**Comment by frnmst** [comment]
Ok Sispheor, 

I don't know how much time I can spend on this but I'll have a go at it. So as long there is no forthcoming deadline, it's ok...

> implement your lib in a dedicated folder called sdk inside Kalliope. Verify that you can import it once Kalliope is reinstalled via setup.py.

Do you mean something like:

    $ git clone https://github.com/kalliope-project/kalliope.git
    $ git clone https://github.com/frnmst/kalliope-rest.git
    $ cd kalliope
    $ mkdir sdk
    $ cp -aR ~/kalliope-rest/* sdk/
    $ python
    >>> from kalliope import sdk

> implement the CLI by adding a flag into the current kalliope entry point. Like kalliope cli <command>

Editing the `main` function in `kalliope/__init__.py` so that it works (for example) like the following:

    $ kalliope cli exec by-audio "hello-command.wav" --voice

**Comment by Sispheor** [comment]
Yes exactly.
And also remove the .git folder from your repo. To follow your files from the main kalliope git repo only.

**Comment by frnmst** [comment]
Hello,
I haven't made much progress with kalliope-rest because I am continuing with [kalliope-docker](https://github.com/frnmst/kalliope-docker/tree/dev) first.
I hope I can get on with kalliope-rest soon, once kalliope-docker is ready.
